test/e2e/data/e2e_conf.yaml

```diff
@@ -71,17 +107,21 @@ providers:
 - name: openstack
   type: InfrastructureProvider
   versions:
+  # Note: for clusterctl upgrade tests, we add lower versions programmatically here, e.g. v0.5.3
   - name: v0.6.99
     value: ../../../config/default
-    contract: v1beta1
+    # This is the upcoming version.
+    # Specify no contract so that upgrade tests that start from a specific contract won't pick it up.
+    # contract: v1beta1
     files:
     - sourcePath: "../data/shared/v1beta1_provider/metadata.yaml"
     - sourcePath: "./infrastructure-openstack/cluster-template.yaml"
     - sourcePath: "./infrastructure-openstack/cluster-template-external-cloud-provider.yaml"
     - sourcePath: "./infrastructure-openstack/cluster-template-without-lb.yaml"
     replacements:
+    # TODO(lentzi90): We should use the e2e tag here instead of main
     - old: gcr.io/k8s-staging-capi-openstack/capi-openstack-controller:dev
-      new: gcr.io/k8s-staging-capi-openstack/capi-openstack-controller:e2e
+      new: gcr.io/k8s-staging-capi-openstack/capi-openstack-controller:main
```

Comment:
is there any reason we didn't use e2e with current PR?

Reply:
It is just to see that the test works. I could not get it to properly import the e2e image for some reason so testing with this until I have solved it